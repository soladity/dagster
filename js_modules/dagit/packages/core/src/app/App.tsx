import * as React from 'react';
import styled from 'styled-components/macro';

import {LeftNav, LEFT_NAV_WIDTH} from '../nav/LeftNav';

import {LayoutContext} from './LayoutProvider';

export const App: React.FC = (props) => {
  const {nav} = React.useContext(LayoutContext);

  const onClickMain = React.useCallback(() => {
    nav.close();
  }, [nav]);

  return (
    <Container>
      <LeftNav />
      <Main $navOpen={nav.isOpen} onClick={onClickMain}>
        {props.children}
      </Main>
    </Container>
  );
};

const Main = styled.div<{$navOpen: boolean}>`
  height: 100%;
  margin-left: ${LEFT_NAV_WIDTH}px;
  width: calc(100% - ${LEFT_NAV_WIDTH}px);

  @media (max-width: 1440px) {
    margin-left: 0;
    width: 100%;
  }
`;

const Container = styled.div`
  display: flex;
  height: calc(100% - 64px);
`;
