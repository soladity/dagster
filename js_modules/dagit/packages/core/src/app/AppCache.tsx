import {InMemoryCache, defaultDataIdFromObject} from '@apollo/client';

// this is a require cause otherwise it breaks
import possibleTypes from '../graphql/possibleTypes.generated.json';

export const createAppCache = () =>
  new InMemoryCache({
    addTypename: true,
    possibleTypes,
    dataIdFromObject: (object: any) => {
      if (
        object.name &&
        (object.__typename === 'RegularType' || object.__typename === 'CompositeType')
      ) {
        return `Type.${object.name}`;
      } else if (object.__typename === 'Instance') {
        return 'Instance';
      } else if (object.__typename === 'Workspace') {
        return 'Workspace';
      } else {
        return defaultDataIdFromObject(object);
      }
    },
  });
