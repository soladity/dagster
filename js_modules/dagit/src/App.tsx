import * as React from "react";

import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";

import CustomAlertProvider from "./CustomAlertProvider";
import { NonIdealState } from "@blueprintjs/core";
import { PipelineExecutionRoot } from "./execute/PipelineExecutionRoot";
import { PipelineExecutionSetupRoot } from "./execute/PipelineExecutionSetupRoot";
import PipelineExplorerRoot from "./PipelineExplorerRoot";
import PythonErrorInfo from "./PythonErrorInfo";
import { RootPipelinesQuery } from "./types/RootPipelinesQuery";
import { RunRoot } from "./runs/RunRoot";
import { RunsRoot } from "./runs/RunsRoot";
import { SolidsRoot } from "./solids/SolidsRoot";
import SchedulesRoot from "./schedules/SchedulesRoot";
import { ScheduleRoot } from "./schedules/ScheduleRoot";
import { TopNav } from "./TopNav";
import gql from "graphql-tag";
import { useQuery } from "react-apollo";

function extractData(result?: RootPipelinesQuery) {
  if (!result || !result.pipelinesOrError) {
    return { pipelines: [], error: null };
  }
  if (result.pipelinesOrError.__typename === "PipelineConnection") {
    return { pipelines: result.pipelinesOrError.nodes, error: null };
  } else {
    return { pipelines: [], error: result.pipelinesOrError };
  }
}

const AppRoutes = () => (
  <Switch>
    <Route path="/runs/:runId" component={RunRoot} />
    <Route path="/runs" component={RunsRoot} exact={true} />
    <Route path="/solids/:name?" component={SolidsRoot} />
    <Route path="/p/:pipelineName/runs/:runId" component={RunRoot} />
    <Redirect
      from="/p/:pipelineName"
      exact={true}
      to="/p/:pipelineName/explore"
    />

    <Route
      path="/p/:pipelineName/execute/setup"
      component={PipelineExecutionSetupRoot}
    />
    <Route path="/p/:pipelineName/execute" component={PipelineExecutionRoot} />
    {/* Capture solid subpath in a regex match */}
    <Route
      path="/p/:pipelineName/explore(/?.*)"
      component={PipelineExplorerRoot}
    />
    {/* Legacy redirects */}
    <Redirect
      from="/execute/:pipelineName/setup"
      to="/p/:pipelineName/execute/setup"
    />
    <Redirect from="/execute/:pipelineName" to="/p/:pipelineName/execute" />
    <Redirect
      from="/explore/:pipelineName/:rest?"
      to="/p/:pipelineName/explore/:rest?"
    />
    {/* Index default */}
    <Route path="/scheduler/:rest?" component={SchedulesRoot} />
    <Route path="/schedule/:scheduleName" component={ScheduleRoot} />
    <Route
      render={() => (
        <NonIdealState
          title="No pipeline selected"
          description="Select a pipeline in the navbar"
        />
      )}
    />
  </Switch>
);

export const App: React.FunctionComponent = () => {
  const result = useQuery<RootPipelinesQuery>(ROOT_PIPELINES_QUERY, {
    fetchPolicy: "cache-and-network"
  });
  const { pipelines, error } = extractData(result.data);

  return (
    <BrowserRouter>
      <TopNav pipelines={pipelines} />
      {error ? (
        <PythonErrorInfo
          contextMsg={`${error.__typename} encountered when loading pipelines:`}
          error={error}
          centered={true}
        />
      ) : (
        <AppRoutes />
      )}
      <CustomAlertProvider />
    </BrowserRouter>
  );
};

export const ROOT_PIPELINES_QUERY = gql`
  query RootPipelinesQuery {
    pipelinesOrError {
      __typename
      ... on PythonError {
        message
        stack
      }
      ... on PipelineConnection {
        nodes {
          ...TopNavPipelinesFragment
        }
      }
    }
  }

  ${TopNav.fragments.TopNavPipelinesFragment}
`;
