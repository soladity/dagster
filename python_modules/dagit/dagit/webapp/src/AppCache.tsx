import {
  InMemoryCache,
  IntrospectionFragmentMatcher,
  defaultDataIdFromObject
} from "apollo-cache-inmemory";
// this is a require cause otherwise it breaks
const introspectionQueryResultData = require("./schema.json");

const fragmentMatcher = new IntrospectionFragmentMatcher({
  introspectionQueryResultData: {
    __schema: introspectionQueryResultData
  }
});

const AppCache = new InMemoryCache({
  fragmentMatcher,
  cacheRedirects: {
    Query: {
      pipeline: (_, args, { getCacheKey }) =>
        getCacheKey({ __typename: "Pipeline", name: args.name })
    }
  },
  dataIdFromObject: (object: any) => {
    if (object.__typename === "Pipeline" && object.name) {
      return `Pipeline.${object.name}`;
    } else {
      return defaultDataIdFromObject(object);
    }
  }
});

export default AppCache;
