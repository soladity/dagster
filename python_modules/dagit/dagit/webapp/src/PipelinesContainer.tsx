import * as React from "react";
import gql from "graphql-tag";
import { Query, QueryResult } from "react-apollo";
import { History } from "history";
import Loading from "./Loading";
import Pipelines from "./Pipelines";
import { PipelinesContainerQuery } from "./types/PipelinesContainerQuery";

interface IPipelinesContainerProps {
  pipelineName: string;
  history: History;
}

export default class PipelinesContainer extends React.Component<
  IPipelinesContainerProps,
  {}
> {
  render() {
    return (
      <Query query={PIPELINES_CONTAINER_QUERY}>
        {(queryResult: QueryResult<PipelinesContainerQuery, any>) => {
          return (
            <Loading queryResult={queryResult}>
              {data => {
                return (
                  <Pipelines
                    selectedPipeline={this.props.pipelineName}
                    pipelines={data.pipelines}
                    history={this.props.history}
                  />
                );
              }}
            </Loading>
          );
        }}
      </Query>
    );
  }
}

export const PIPELINES_CONTAINER_QUERY = gql`
  query PipelinesContainerQuery {
    pipelines {
      ...PipelinesFragment
    }
  }

  ${Pipelines.fragments.PipelinesFragment}
`;
