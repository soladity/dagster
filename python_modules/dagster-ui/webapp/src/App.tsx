import * as React from "react";
import gql from "graphql-tag";
import { Query, QueryResult } from "react-apollo";
import { Route } from "react-router";
import { BrowserRouter } from "react-router-dom";
import Page from "./Page";
import Loading from "./Loading";
import Pipelines from "./Pipelines";
import { AppQuery } from "./types/AppQuery";

class App extends React.Component {
  public render() {
    return (
      <BrowserRouter>
        <Route path="/:pipeline">
          {({ match, history }) => (
            <Query query={APP_QUERY}>
              {(queryResult: QueryResult<AppQuery, any>) => (
                <Page>
                  <Loading queryResult={queryResult}>
                    {(data: AppQuery) => (
                      <Pipelines
                        selectedPipeline={match ? match.params.pipeline : ""}
                        pipelines={data.pipelines}
                        history={history}
                      />
                    )}
                  </Loading>
                </Page>
              )}
            </Query>
          )}
        </Route>
      </BrowserRouter>
    );
  }
}

const APP_QUERY = gql`
  query AppQuery {
    pipelines {
      ...PipelinesFragment
    }
  }

  ${Pipelines.fragments.PipelinesFragment}
`;

export default App;
