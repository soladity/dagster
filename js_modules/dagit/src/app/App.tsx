import * as React from 'react';
import {BrowserRouter, Route, Switch} from 'react-router-dom';

import {CustomAlertProvider} from 'src/app/CustomAlertProvider';
import {CustomConfirmationProvider} from 'src/app/CustomConfirmationProvider';
import {CustomTooltipProvider} from 'src/app/CustomTooltipProvider';
import {APP_PATH_PREFIX} from 'src/app/DomUtils';
import {FallthroughRoot} from 'src/app/FallthroughRoot';
import {FeatureFlagsRoot} from 'src/app/FeatureFlagsRoot';
import {SettingsRoot} from 'src/app/SettingsRoot';
import {TimezoneProvider} from 'src/app/time/TimezoneContext';
import {useDocumentTitle} from 'src/hooks/useDocumentTitle';
import {InstanceRoot} from 'src/instance/InstanceRoot';
import {LeftNavBranch} from 'src/nav/LeftNavBranch';
import {AllSchedulesRoot} from 'src/schedules/AllSchedulesRoot';
import {AllSensorsRoot} from 'src/sensors/AllSensorsRoot';
import {WorkspaceProvider} from 'src/workspace/WorkspaceContext';
import {WorkspaceRoot} from 'src/workspace/WorkspaceRoot';

export const AppContent = () => {
  useDocumentTitle('Dagit');
  return (
    <div style={{display: 'flex', height: '100%'}}>
      <WorkspaceProvider>
        <LeftNavBranch />
        <CustomConfirmationProvider>
          <Switch>
            <Route path="/flags" component={FeatureFlagsRoot} />
            <Route path="/instance" component={InstanceRoot} />
            <Route path="/workspace" component={WorkspaceRoot} />
            <Route path="/schedules" component={AllSchedulesRoot} />
            <Route path="/sensors" component={AllSensorsRoot} />
            <Route path="/settings" component={SettingsRoot} />
            <Route path="*" component={FallthroughRoot} />
          </Switch>
          <CustomTooltipProvider />
          <CustomAlertProvider />
        </CustomConfirmationProvider>
      </WorkspaceProvider>
    </div>
  );
};

export const App = () => (
  <BrowserRouter basename={APP_PATH_PREFIX}>
    <TimezoneProvider>
      <AppContent />
    </TimezoneProvider>
  </BrowserRouter>
);
