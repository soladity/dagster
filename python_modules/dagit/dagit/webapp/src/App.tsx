import * as React from "react";
import { Switch, Route } from "react-router";
import { BrowserRouter } from "react-router-dom";
import Page from "./Page";
import PipelinesContainer from "./PipelinesContainer";

export default class App extends React.Component {
  public render() {
    return (
      <Page>
        <BrowserRouter>
          <Route
            path="/:pipeline?"
            render={({ match, history }) => {
              return (
                <PipelinesContainer
                  pipelineName={match.params.pipeline || ""}
                  history={history}
                />
              );
            }}
          />
        </BrowserRouter>
      </Page>
    );
  }
}
