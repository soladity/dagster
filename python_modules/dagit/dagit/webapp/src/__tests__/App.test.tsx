import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import App from "../App";
import { PIPELINES_CONTAINER_QUERY } from "../PipelinesContainer";
import AppCache from "../AppCache";
import { MockedProvider } from "./MockedProvider";

const mocks = [
  {
    request: {
      operationName: "PipelineseContainerQuery",
      query: PIPELINES_CONTAINER_QUERY
    },
    result: {
      data: {
        pipelines: [
          {
            name: "pandas_hello_world",
            description: null,
            solids: [
              {
                outputs: [
                  {
                    definition: {
                      name: "result",
                      type: {
                        name: "PandasDataFrame",
                        description:
                          "Two-dimensional size-mutable, potentially heterogeneous\ntabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/",
                        __typename: "RegularType"
                      },
                      __typename: "OutputDefinition",
                      description: null,
                      expectations: []
                    },
                    __typename: "Output"
                  }
                ],
                inputs: [
                  {
                    definition: {
                      name: "sum_df",
                      type: {
                        name: "PandasDataFrame",
                        description:
                          "Two-dimensional size-mutable, potentially heterogeneous\ntabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/",
                        __typename: "RegularType"
                      },
                      __typename: "InputDefinition",
                      description: null,
                      expectations: []
                    },
                    __typename: "Input",
                    dependsOn: {
                      definition: {
                        name: "result",
                        __typename: "OutputDefinition"
                      },
                      solid: { name: "sum_solid", __typename: "Solid" },
                      __typename: "Output"
                    }
                  }
                ],
                __typename: "Solid",
                name: "sum_sq_solid",
                definition: {
                  description: null,
                  configDefinition: {
                    type: {
                      __typename: "RegularType",
                      name: "Any",
                      description:
                        "The type that allows any value, including no value."
                    },
                    __typename: "Config"
                  },
                  __typename: "SolidDefinition"
                }
              },
              {
                outputs: [
                  {
                    definition: {
                      name: "result",
                      type: {
                        name: "PandasDataFrame",
                        description:
                          "Two-dimensional size-mutable, potentially heterogeneous\ntabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/",
                        __typename: "RegularType"
                      },
                      __typename: "OutputDefinition",
                      description: null,
                      expectations: []
                    },
                    __typename: "Output"
                  }
                ],
                inputs: [],
                __typename: "Solid",
                name: "load_num_csv",
                definition: {
                  description: null,
                  configDefinition: {
                    type: {
                      __typename: "CompositeType",
                      name: "ConfigDictionary",
                      description:
                        "Configuration dictionary.\n\n    Typed-checked but then passed to implementations as a python dict\n\n    Arguments:\n      fields (dict): dictonary of :py:class:`Field` objects keyed by name",
                      fields: [
                        {
                          name: "path",
                          description: null,
                          isOptional: false,
                          defaultValue:
                            "<class 'dagster.core.types.__FieldValueSentinel'>",
                          type: {
                            name: "Path",
                            description:
                              "\nA string the represents a path. It is very useful for some tooling\nto know that a string indeed represents a file path. That way they\ncan, for example, make the paths relative to a different location\nfor a particular execution environment.\n",
                            __typename: "RegularType"
                          },
                          __typename: "TypeField"
                        }
                      ]
                    },
                    __typename: "Config"
                  },
                  __typename: "SolidDefinition"
                }
              },
              {
                outputs: [
                  {
                    definition: {
                      name: "result",
                      type: {
                        name: "PandasDataFrame",
                        description:
                          "Two-dimensional size-mutable, potentially heterogeneous\ntabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/",
                        __typename: "RegularType"
                      },
                      __typename: "OutputDefinition",
                      description: null,
                      expectations: []
                    },
                    __typename: "Output"
                  }
                ],
                inputs: [
                  {
                    definition: {
                      name: "num",
                      type: {
                        name: "PandasDataFrame",
                        description:
                          "Two-dimensional size-mutable, potentially heterogeneous\ntabular data structure with labeled axes (rows and columns). See http://pandas.pydata.org/",
                        __typename: "RegularType"
                      },
                      __typename: "InputDefinition",
                      description: null,
                      expectations: []
                    },
                    __typename: "Input",
                    dependsOn: {
                      definition: {
                        name: "result",
                        __typename: "OutputDefinition"
                      },
                      solid: { name: "load_num_csv", __typename: "Solid" },
                      __typename: "Output"
                    }
                  }
                ],
                __typename: "Solid",
                name: "sum_solid",
                definition: {
                  description: null,
                  configDefinition: {
                    type: {
                      __typename: "RegularType",
                      name: "Any",
                      description:
                        "The type that allows any value, including no value."
                    },
                    __typename: "Config"
                  },
                  __typename: "SolidDefinition"
                }
              }
            ],
            contexts: [
              {
                name: "default",
                description: null,
                config: {
                  type: {
                    __typename: "CompositeType",
                    name: "ConfigDictionary",
                    description:
                      "Configuration dictionary.\n\n    Typed-checked but then passed to implementations as a python dict\n\n    Arguments:\n      fields (dict): dictonary of :py:class:`Field` objects keyed by name",
                    fields: [
                      {
                        name: "log_level",
                        description: null,
                        isOptional: true,
                        defaultValue: "INFO",
                        type: {
                          name: "String",
                          description: "A string.",
                          __typename: "RegularType"
                        },
                        __typename: "TypeField"
                      }
                    ]
                  },
                  __typename: "Config"
                },
                __typename: "PipelineContext"
              }
            ],
            __typename: "Pipeline"
          }
        ]
      }
    }
  }
];

function createNodeMock(element: any) {
  if (element.type === "div") {
    return {
      querySelector() {
        return null;
      },
      querySelectorAll() {
        return [];
      }
    };
  }
  return null;
}

it("renders without error", async () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={mocks} addTypename={true} cache={AppCache}>
      <App />
    </MockedProvider>,
    { createNodeMock }
  );

  await new Promise(resolve => setTimeout(resolve, 1000));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
