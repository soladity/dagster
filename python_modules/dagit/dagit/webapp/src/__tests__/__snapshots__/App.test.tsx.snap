// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `
Array [
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
      >
        <img
          style={
            Object {
              "height": 34,
            }
          }
        />
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <div
        className="sc-brqgnP jyxKUE"
      >
        <span
          className="bp3-popover-wrapper"
        >
          <span
            className="bp3-popover-target"
            onClick={[Function]}
          >
            <div
              className=""
              onKeyDown={[Function]}
            >
              <button
                className="bp3-button"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                <span
                  className="bp3-button-text"
                >
                  Select a Pipeline...
                </span>
                <svg
                  className="bp3-icon"
                  data-icon="double-caret-vertical"
                  height={16}
                  style={Object {}}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </span>
        </span>
        <div
          className="sc-cMljjf bKfJvu"
        >
          /
        </div>
      </div>
    </div>
  </div>,
  <div
    className="bp3-non-ideal-state"
  >
    <h4
      className="bp3-heading"
    >
      No pipeline selected
    </h4>
    <div>
      Select a pipeline in the sidebar on the left
    </div>
  </div>,
]
`;
